Last-Update: 2012-09-13
Author: Dmitry Smirnov <onlyjob@member.fsf.org>
Forwarded: not-needed
Description: disable building bundled cppconn
 To use libmysqlcppconn-dev (>= 1.1.1) instead.
 (previous versions have no method 'isNumeric').

--- a/configure.in
+++ b/configure.in
@@ -318,6 +318,13 @@
 #AC_CHECK_LIB(GL, glNewList, CAIRO_LIBS="$CAIRO_LIBS -lGL", AC_MSG_ERROR([OpenGL library not found]))
 
 
+# check for libmysqlcppconn
+AC_CHECK_HEADER(cppconn/connection.h, [], AC_MSG_ERROR([cppconn headers not found]))
+CPPCONN_CFLAGS="-I${includedir}/cppconn"
+CPPCONN_LIBS="-lmysqlcppconn"
+AC_SUBST(CPPCONN_CFLAGS)
+AC_SUBST(CPPCONN_LIBS)
+
 # check if ctemplate is installed in system, otherwise use the bundled one
 AC_CHECK_HEADER(ctemplate/template.h, ac_system_ctemplate=yes, ac_system_ctemplate=no)
 
@@ -492,8 +499,6 @@
 res/sqlidedata/Makefile
 ext/Makefile
 ext/scintilla/gtk/Makefile
-ext/cppconn/Makefile
-ext/cppconn/driver/Makefile
 ext/vsqlite++/Makefile
 ext/vsqlite++/src/sqlite/Makefile
 backend/Makefile
--- a/backend/wbprivate/Makefile.am
+++ b/backend/wbprivate/Makefile.am
@@ -10,7 +10,8 @@
   -I$(top_srcdir)/backend/wbpublic -I$(top_srcdir)/generated -I$(top_srcdir)/library/canvas/src\
   -I$(top_srcdir)/backend/wbprivate/workbench \
   -I$(top_srcdir)/backend/wbpublic/grt \
-  -I$(top_srcdir)/ext/cppconn/ -I$(top_srcdir)/library/dbc/src\
+  @CPPCONN_CFLAGS@ \
+  -I$(top_srcdir)/library/dbc/src\
   -I$(top_srcdir)/ext/scintilla/include\
   -I$(top_srcdir)/ext/vsqlite++/include\
   -I$(top_srcdir)/ext/vsqlite\
--- a/frontend/linux/sqlide/Makefile.am
+++ b/frontend/linux/sqlide/Makefile.am
@@ -15,7 +15,7 @@
 -I$(top_srcdir)/library/grt/src\
 -I$(top_srcdir)/library/base\
 -I$(top_srcdir)/library/dbc/src\
--I$(top_srcdir)/ext/cppconn\
+@CPPCONN_CFLAGS@\
 -I$(top_srcdir)/ext/vsqlite++/include\
 -I$(top_srcdir)/ext/scintilla/include\
 @GRT_CFLAGS@ @CAIRO_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@ @GNOME_CFLAGS@
--- a/frontend/linux/workbench/Makefile.am
+++ b/frontend/linux/workbench/Makefile.am
@@ -38,7 +38,7 @@
 	-I$(top_srcdir)/modules/wb.model/src\
 	-I$(top_srcdir)/modules/interfaces\
 	-I$(top_srcdir)/ext\
-	-I$(top_srcdir)/ext/cppconn\
+	@CPPCONN_CFLAGS@\
 	-I$(top_srcdir)/modules\
   -I$(top_srcdir)/ext/vsqlite++/include\
   -I$(top_srcdir)/ext/scintilla/include\
@@ -54,7 +54,7 @@
 	$(top_srcdir)/library/base/libwbbase.la\
 	$(top_srcdir)/library/grt/src/libgrt.la\
 	$(top_srcdir)/library/dbc/libcdbc.la\
-	$(top_srcdir)/ext/cppconn/driver/mysqlcppconn.la\
+	@CPPCONN_LIBS@\
 	$(top_srcdir)/ext/scintilla/gtk/libwbscintilla.la\
 	$(top_srcdir)/library/canvas/src/libmdcanvasgtk.la\
 	$(top_srcdir)/library/canvas/src/libmdcanvas.la\
--- a/library/dbc/Makefile.am
+++ b/library/dbc/Makefile.am
@@ -12,6 +12,6 @@
 -I$(top_srcdir)/library/grt/src\
 -I$(top_srcdir)/library/base\
 -I$(top_srcdir)/ext\
--I$(top_srcdir)/ext/cppconn
+@CPPCONN_CFLAGS@
 
 
--- a/modules/db.mysql.query/Makefile.am
+++ b/modules/db.mysql.query/Makefile.am
@@ -6,7 +6,7 @@
 INCLUDES=-I$(top_srcdir)/generated -I$(top_srcdir)/library/grt/src \
   -I$(top_srcdir)/library/base\
   -I$(top_srcdir)/library/dbc/src\
-  -I$(top_srcdir)/ext/cppconn\
+  @CPPCONN_CFLAGS@\
   @GRT_CFLAGS@ @GLIB_CFLAGS@ @SIGC_CFLAGS@
 
 
--- a/plugins/db.mysql.diff.reporting/frontend/Makefile.am
+++ b/plugins/db.mysql.diff.reporting/frontend/Makefile.am
@@ -11,7 +11,7 @@
     -I$(top_srcdir)/ext/scintilla/include\
     -I$(top_srcdir)/modules/db.mysql/src\
     -I$(top_srcdir)/library/dbc/src\
-    -I$(top_srcdir)/ext/cppconn\
+    @CPPCONN_CFLAGS@\
     -I$(top_srcdir)/modules/interfaces\
     -I$(top_srcdir)/modules/\
     -I$(top_srcdir)/library/grt/src/diff\
--- a/plugins/db.mysql/frontend/Makefile.am
+++ b/plugins/db.mysql/frontend/Makefile.am
@@ -10,7 +10,7 @@
     -I$(top_srcdir)/modules/db.mysql/src\
     -I$(top_srcdir)/library/dbc/src\
     -I$(top_srcdir)/library/forms/mforms\
-    -I$(top_srcdir)/ext/cppconn\
+    @CPPCONN_CFLAGS@\
     -I$(top_srcdir)/modules/interfaces\
     -I$(top_srcdir)/modules/\
     -I$(top_srcdir)/library/grt/src/diff\
--- a/plugins/wb.model.snippets/Makefile.am
+++ b/plugins/wb.model.snippets/Makefile.am
@@ -11,7 +11,7 @@
     -I$(top_srcdir)/ext/scintilla/include\
     -I$(top_srcdir)/modules/db.mysql/src\
     -I$(top_srcdir)/library/dbc/src\
-    -I$(top_srcdir)/ext/cppconn\
+    @CPPCONN_CFLAGS@\
     -I$(top_srcdir)/modules/interfaces\
     -I$(top_srcdir)/modules/\
     -I$(top_srcdir)/library/grt/src/diff\
--- a/backend/wbprivate/sqlide/wb_sql_editor_form.cpp
+++ b/backend/wbprivate/sqlide/wb_sql_editor_form.cpp
@@ -49,7 +49,7 @@
 // this doesn't belong here, but there's no other workaround for having access to mysql_info
 // because of that we also need to link wbprivate directly to the connector, making their
 // dynamic loading pretty much useless
-#include <cppconn/../driver/mysql_connection.h>
+#include <mysql_connection.h>
 
 #include <boost/foreach.hpp>
 #include <boost/scoped_ptr.hpp>
--- a/ext/Makefile.am
+++ b/ext/Makefile.am
@@ -1,5 +1,5 @@
 
-SUBDIRS=scintilla/gtk cppconn vsqlite++ antlr-runtime
+SUBDIRS=scintilla/gtk vsqlite++ antlr-runtime
 
 .PHONY: install-utils
 
--- a/backend/wbpublic/Makefile.am
+++ b/backend/wbpublic/Makefile.am
@@ -15,7 +15,7 @@
 	-I$(top_srcdir)/library/forms\
 	-I$(top_srcdir)/library/forms/mforms/\
 	-I$(top_srcdir)/library/canvas/src\
-	-I$(top_srcdir)/ext/cppconn\
+	@CPPCONN_CFLAGS@\
 	-I$(top_srcdir)/ext/scintilla/include\
 	-I$(top_srcdir)/library/dbc/src\
 	-I$(top_srcdir)/library/mysql55-parser\
@@ -167,4 +167,4 @@
 #stdafx.h.gch: stdafx.h
 #	$(CXXCOMPILE) $<
 
-libwbpublic_la_LIBADD=../../library/canvas/src/libmdcanvas.la ../../library/forms/libmforms.la ../../library/dbc/libcdbc.la ../../ext/vsqlite++/src/sqlite/libvsqlitepp.la ../../ext/scintilla/gtk/libwbscintilla.la @CTEMPLATE_LIBS@
+libwbpublic_la_LIBADD=../../library/canvas/src/libmdcanvas.la ../../library/forms/libmforms.la ../../library/dbc/libcdbc.la ../../ext/vsqlite++/src/sqlite/libvsqlitepp.la ../../ext/scintilla/gtk/libwbscintilla.la @CTEMPLATE_LIBS@ @CPPCONN_LIBS@
--- a/backend/wbpublic/grtdb/diff_dbobjectmatch.cpp
+++ b/backend/wbpublic/grtdb/diff_dbobjectmatch.cpp
@@ -22,7 +22,7 @@
 // this doesn't belong here, but there's no other workaround for having access to mysql_info
 // because of that we also need to link wbprivate directly to the connector, making their
 // dynamic loading pretty much useless
-#include <cppconn/../driver/mysql_connection.h>
+#include <mysql_connection.h>
 #include <cppconn/metadata.h>
 
 
